stl/inc/iterator

```diff
@@ -513,6 +513,501 @@ private:
+            } else {
+                return _Left._Val._Iterator == _Right._Val._Sentinel;
+            }
+        } else {
+            if (_Right._Val._Contains == _Variantish_state::_Holds_iter) {
+                return _Left._Val._Sentinel == _Right._Val._Iterator;
+            } else {
+                return true;
+            }
+        }
+    }
+
+    // clang-format off
+    template <sized_sentinel_for<_Iter> _OIter, sized_sentinel_for<_Iter> _OSe>
+        requires sized_sentinel_for<_Se, _OIter>
+    _NODISCARD friend iter_difference_t<_OIter> operator-(
+        const common_iterator& _Left, const common_iterator<_OIter, _OSe>& _Right) {
+        // clang-format on
+#if _ITERATOR_DEBUG_LEVEL != 0
+        _STL_VERIFY(
+            _Left._Val._Contains != _Variantish_state::_Nothing && _Right._Val._Contains != _Variantish_state::_Nothing,
+            "Cannot take difference of valueless common_iterators");
+#endif // _ITERATOR_DEBUG_LEVEL != 0
+
+        if (_Left._Val._Contains == _Variantish_state::_Holds_iter) {
+            if (_Right._Val._Contains == _Variantish_state::_Holds_iter) {
+                return _Left._Val._Iterator - _Right._Val._Iterator;
+            } else {
+                return _Left._Val._Iterator - _Right._Val._Sentinel;
+            }
+        } else {
+            if (_Right._Val._Contains == _Variantish_state::_Holds_iter) {
+                return _Left._Val._Sentinel - _Right._Val._Iterator;
+            } else {
+                return 0;
+            }
+        }
+    }
+
+    _NODISCARD friend iter_rvalue_reference_t<_Iter> iter_move(const common_iterator& _Right) noexcept(
+        noexcept(_RANGES iter_move(_Right._Val._Iterator))) requires input_iterator<_Iter> {
+#if _ITERATOR_DEBUG_LEVEL != 0
+        _STL_VERIFY(_Right._Val._Contains == _Variantish_state::_Holds_iter,
+            "can only iter_move from common_iterator if it holds an iterator");
+#endif // _ITERATOR_DEBUG_LEVEL != 0
+        return _RANGES iter_move(_Right._Val._Iterator);
+    }
+
+    template <indirectly_swappable<_Iter> _OIter, class _OSe>
+    friend void iter_swap(const common_iterator& _Left, const common_iterator<_OIter, _OSe>& _Right) noexcept(
+        noexcept(_RANGES iter_swap(_Left._Val._Iterator, _Right._Val._Iterator))) {
+#if _ITERATOR_DEBUG_LEVEL != 0
+        _STL_VERIFY(_Left._Val._Contains == _Variantish_state::_Holds_iter
+                        && _Right._Val._Contains == _Variantish_state::_Holds_iter,
+            "can only iter_swap common_iterators if both hold iterators");
+#endif // _ITERATOR_DEBUG_LEVEL != 0
+        return _RANGES iter_swap(_Left._Val._Iterator, _Right._Val._Iterator);
+    }
+
+private:
+    // clang-format off
+    template <input_or_output_iterator _OIter, sentinel_for<_OIter> _OSe>
+        requires (!same_as<_OIter, _OSe> && copyable<_OIter>)
+    friend class common_iterator;
+    // clang-format on
+
+    using _Vty = _Variantish<_Iter, _Se>;
+    _Vty _Val;

```

Comment:
This `_Vty` typedef appears to be used nowhere else - is it necessary?

Reply:
No, not since @miscco moved the enumeration out of `_Vty` into namespace scope.